doc/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/index.md

```diff
@@ -221,33 +221,66 @@ Key differences between contributing source code and documentation are :
 
 ## Documentation preview
 
-* Install [Jekyll](https://jekyllrb.com/).
-See [Jekyll Installation page](https://jekyllrb.com/docs/installation/) for detail instructions.
-* Install [jekyll-sitemap](https://github.com/jekyll/jekyll-sitemap).
-Jekyll-sitemap can be installed using [RubyGems](https://rubygems.org/pages/download).
+1. Install required software :
+  * [Git](https://git-scm.com/)
+  * [Ruby](https://www.ruby-lang.org)
+  * [Jekyll](https://jekyllrb.com/)
+  * [Jekyll-sitemap](https://github.com/jekyll/jekyll-sitemap)
+  * [Jekyll-gist](https://github.com/jekyll/jekyll-gist)
+  * [Jekyll-paginate](https://github.com/jekyll/jekyll-paginate)
+  * [Rouge](https://github.com/jneen/rouge)
 
-  ```bash
-gem install jekyll-sitemap
-```
-* Install [rouge](https://github.com/jneen/rouge) for syntax highlighting.
-Rouge can be installed using [RubyGems](https://rubygems.org/pages/download).
+   Detailed installation instructions for selected platforms:
+<ul>
+<li style="list-style-type: none;">
+<ul class="nav nav-tabs">
+  <li class="active"><a data-toggle="tab" href="#Platform1">Ubuntu 14.04</a></li>
+  <li><a data-toggle="tab" href="#Platform2">Ubuntu 16.04</a></li>
+  <li><a data-toggle="tab" href="#Platform3">CentOS 7</a></li>
+</ul>
+
+<div class="tab-content">
+<div id="Platform1" class="tab-pane fade in active" markdown="1" >
 
-  ```bash
-gem install -N rouge
+```bash
+sudo apt-get update
+sudo apt-get install -y git ruby2.0 ruby2.0-dev build-essential
+sudo gem2.0 install -N jekyll jekyll-gist jekyll-sitema rouge
 ```
 
-* Create local git branch gh-pages-stub with same content as origin/gh-pages-stub .
+</div><div id="Platform2" class="tab-pane fade" markdown="1" >
 
-  ```bash
-git fetch origin gh-pages-stub:gh-pages-stub
+```bash
+sudo apt-get update
+sudo apt-get install -y git build-essential ruby ruby-dev ruby-ffi
+sudo gem install -N jekyll jekyll-gist jekyll-sitemap jekyll-paginate rouge
 ```
 
-* Delete `test-gh-pages-current/` if it exists.
-* Run `test-gh-pages.sh` script to generate documentation and start development server at [http://localhost:4000/](http://localhost:4000/).
+</div><div id="Platform3" class="tab-pane fade" markdown="1" >
 
-  ```bash
-user@host:/kaa$ ./test-gh-pages.sh
-...
-Server address: http://127.0.0.1:4000//
-Server running... press ctrl-c to stop.
+```bash
+sudo yum install -y ruby git ruby-devel
+sudo yum groupinstall -y 'Development Tools'
+sudo gem install -N jekyll jekyll-gist jekyll-sitemap jekyll-paginate rouge
 ```
+
+</div></div>
+</li>
+</ul>
+
+{: start="2"}
+2. Create local git branch gh-pages-stub with same content as origin/gh-pages-stub.
+
+   ```bash
+   git fetch origin gh-pages-stub:gh-pages-stub
+   ```
+
+3. Delete `test-gh-pages-current/` if it exists.
+4. Run `test-gh-pages.sh` script to generate documentation and start development server at [http://localhost:4000/](http://localhost:4000/).
+
```

Comment:
@vadimol `git fetch origin gh-pages-stub:gh-pages-stub` doesn't switch to new brounch.